File: src/main/java/com/github/weewar/mapviewer/utils/ClassPath.java
package com.github.weewar.mapviewer.utils;

import com.github.weewar.mapviewer.exceptions.ClassPathResourceException;
import org.springframework.core.io.Resource;
import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
import org.springframework.core.io.support.ResourcePatternResolver;

import java.io.IOException;
import java.net.URL;
import java.util.LinkedList;
import java.util.List;

public class ClassPath {
    private final static ResourcePatternResolver resourceResolver = new PathMatchingResourcePatternResolver();

    public static List<URL> resources(String locationPattern) throws ClassPathResourceException {
        try {
            List<URL> resourceURLs = new LinkedList<>();
            for (Resource resource : resourceResolver.getResources(locationPattern)) {
                resourceURLs.add(resource.getURL());
            }
            return resourceURLs;
        } catch (IOException e) {
            throw new ClassPathResourceException("Unable to locate resources in a classpath: " + locationPattern, e);
        }
    }

    public static URL resource(String path) throws ClassPathResourceException {
        try {
            return resourceResolver.getResource(path).getURL();
        } catch (IOException e) {
            throw new ClassPathResourceException("Unable to locate resource in a classpath: " + path, e);
        }
    }
}

File: src/main/kotlin/com/github/weewar/mapviewer/utils/ClassPath.kt
package com.github.weewar.mapviewer.utils

import com.github.weewar.mapviewer.model.ClassPathResourceException
import org.springframework.core.io.support.PathMatchingResourcePatternResolver
import org.springframework.stereotype.Component
import java.io.IOException
import java.net.URL
import java.util.*

@Component
open class ClassPath {
    private val resourceResolver = PathMatchingResourcePatternResolver()

    @Throws(ClassPathResourceException::class)
    open fun resources(locationPattern: String): List<URL> =
            try {
                val resourceURLs = LinkedList<URL>()
                for (resource in resourceResolver.getResources("$locationPattern*")) {
                    resourceURLs.add(resource.url)
                }
                resourceURLs
            } catch (e: IOException) {
                throw ClassPathResourceException("Unable to locate resources in a classpath: $locationPattern", e)
            }

    @Throws(ClassPathResourceException::class)
    open fun resource(path: String): URL =
            try {
                resourceResolver.getResource(path).url
            } catch (e: IOException) {
                throw ClassPathResourceException("Unable to locate resource in a classpath: $path", e)
            }
}
